Content.Server/Shuttles/Systems/ShuttleTimerSystem.cs

```diff
@@ -0,0 +1,98 @@
+        [Dependency] private readonly StationSystem _stationSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<ShuttleTimerComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+        }
+
+        /// <summary>
+        /// Determines if/how a broadcast packet affects this timer.
+        /// All shuttle timer packets are broadcast in their network, and subnetting is implemented by filtering timer MapUid.
+        /// </summary>
+        private void OnPacketReceived(EntityUid uid, ShuttleTimerComponent component, DeviceNetworkPacketEvent args)
+        {
+            var timerXform = Transform(uid);
+
+            // no false positives.
+            if (timerXform.MapUid == null)
+                return;
+
+            string key;
+            args.Data.TryGetValue("ShuttleMap", out EntityUid? shuttleMap);
+            args.Data.TryGetValue("SourceMap", out EntityUid? source);
+            args.Data.TryGetValue("DestMap", out EntityUid? dest);
+            args.Data.TryGetValue("Docked", out bool docked);
+            string?[] text = new string?[] { docked ? "ETD" : "ETA" };
+
+            switch (timerXform.MapUid)
+            {
+                // sometimes the timer transforms on FTL shuttles have a hyperspace mapuid, so matching by grid works as a fallback.
+                case var local when local == shuttleMap || timerXform.GridUid == shuttleMap:
+                    key = "ShuttleTimer";
+                    break;
+                case var origin when origin == source:
+                    key = "SourceTimer";
+                    break;
+                case var remote when remote == dest:
+                    key = "DestTimer";
+                    text = new string?[] { "ETA" };
+                    break;
+                default:
+                    return;
+            }
+
+            if (!args.Data.TryGetValue(key, out TimeSpan duration))
+                return;
+
+            if (args.Data.TryGetValue("Text", out string?[]? label))
+                text = label;
+
+            _appearanceSystem.SetData(uid, TextScreenVisuals.TargetTime, _gameTiming.CurTime + duration);
+            _appearanceSystem.SetData(uid, TextScreenVisuals.ScreenText, text);
+        }
+
+        /// <summary>
+        /// Helper method for <see cref="RoundEndSystem"/> and <see cref="EmergencyShuttleSystem"/> to get the MapUids of
+        /// the `eshuttle`, `centcomm`, and the best-guess `station` (using <see cref="StationSystem.GetLargestGrid(StationDataComponent)"/>)
+        /// </summary>
+        /// <returns>{"eshuttle":uid?, "centcomm":uid?, "station":uid?}</returns>
+        public Dictionary<string, EntityUid?> GetEscapeMaps()
+        {
+            AllEntityQuery<StationEmergencyShuttleComponent>().MoveNext(out var station, out var eshuttleComp);
+            var targetGrid = _stationSystem.GetLargestGrid(Comp<StationDataComponent>(station));
```

Comment:
You already have the query here just add StationDataComponent as another generic.